build-system/tasks/codecov-upload.js

```diff
@@ -58,7 +63,9 @@ async function codecovUpload() {
     );
     return;
   }
-  const commitSha = shortSha(travisPullRequestSha());
+  const commitSha = shortSha(
+    isTravisPullRequestBuild() ? travisPullRequestSha() : travisCommitSha()
```

Comment:
I think this logic should be contained in `travis.js`.

Reply:
Why do we have different logic? Taking this from `git.js`, does `git rev-parse --verify HEAD` give a different SHA than `process.env.TRAVIS_COMMIT`?

Reply:
I believe `git rev-parse --verify HEAD` returns the SHA of the merge commit sent by Github to Travis for PRs, which is different from `TRAVIS_PULL_REQUEST_SHA`. It's probably the same as `TRAVIS_COMMIT` for push builds. Note that `git.js` uses `git rev-parse` only for local runs and not on Travis.

In general, I agree that there may be scope for consolidating the logic in `git.js` and `travis.js`. Since they're used in different ways by pr check, visual diff, bundle size, code coverage, etc., I hesitate to make any logic changes to existing use cases in this PR. (Case in point: the series of bug fixes to [`git.js`](https://github.com/ampproject/amphtml/commits/master/build-system/git.js) by  @danielrozenberg.) A separate, targeted PR to remove redundancy is welcome though :smiley:

Reply:
Okay SG